cmd/soroban-cli/src/log/event.rs

```diff
@@ -0,0 +1,50 @@
+use crate::xdr;
+
+mod log {
+    pub fn log(i: usize, event: &super::xdr::DiagnosticEvent) {
+        tracing::info!("{i}: {event:#?}");
+    }
+}
+
+mod contract {
+
+    pub fn contract(i: usize, event: &super::xdr::DiagnosticEvent) {
+        tracing::info!("{i}: {event:#?}");
+    }
+}
+
+mod diagnostic {
+    pub fn diagnostic(i: usize, event: &super::xdr::DiagnosticEvent) {
+        tracing::debug!("{i}: {event:#?}");
+    }
+}
+
+pub fn events(events: &[xdr::DiagnosticEvent]) {
+    for (i, event) in events.iter().enumerate() {
+        if is_contract_event(event) {
+            contract::contract(i, event);
+        } else if is_log_event(event) {
+            log::log(i, event);
+        } else {
+            diagnostic::diagnostic(i, event);
```

Comment:
❗ Can the log events be `debug` and the diagnostic events should be `trace`?

Contract log events are only outputted from contracts build with a debug flag enabled, and system logs are similar debug information.

Other diagnostic logs are really trace level containing information about each invoke and return value and costs.

Reply:
Sure that makes sense. My one thought on the logs being debug is that new devs will use the logs and then be confused when they aren't present. So perhaps we should print a message that indicates that the logs are present and require `--verbose` or `RUST_LOGS=soroban_cli::log::event::log=info` or `--filter-logs ...`. `verbose` will generate more output that they probably want. And the other solution is a bit wordy. We could also add a `SOROBAN_LOGS` env that might be better?

Reply:
> My one thought on the logs being debug is that new devs will use the logs and then be confused when they aren't present.

That's a good point. The problem is that entries matching the `is_log_event` function here include system logs, not just contract logs, so we're pushing logs that are system related into info too. Leave the logs as info and we'll see if we tweak it over time.